### DISCUSSION TITLE
External Login - Oauth2 Logout Issue

### DISCUSSION THREAD

**Discussion by Sharafudeen** [oqtane/oqtane.framework#3350] [General] [answered]
Hi,

I've enabled oAuth2 External login and configured ("Ping Federate" one of the identity provider) for the root site. Login works perfectly. When I'm trying to logout and it seems logged out but it is not actually logging out. I've checked by refreshing the pages and it shows logout buttons and other authorized options such as menu link and so on. This might be an issue on "Not Terminating External Login Provider Session" such as cookies or logged in user info?

**Additional Info:** PingID team says, You have to call logout Url (provided Url by PingID team) for terminating the logged in user. But I could not find any options to configure logout Url.

**Comment by sbwalker** [comment, marked as answer]
@Sharafudeen Oqtane allows you to logout locally - which removes your local auth cookie and requires you to login once again in order to access secure features. The Oqtane OIDC  integration currently does not remove the auth cookies for the remote IDP used for external login. This is because the OIDC protocol does not include strict standards around the logout scenario and every IDP implements it in their own custom way. So as a result, it is not possible to "productize" the configuration in the same way that login is supported. If the local logout is not sufficient for your needs then you will need to build your own logout button which handles the integration with the specific IDP you are using.
